include/zephyr/drivers/gpio.h

```diff
@@ -305,7 +305,9 @@ struct gpio_dt_spec {
  * @brief Static initializer for a @p gpio_dt_spec
  *
  * This returns a static initializer for a @p gpio_dt_spec structure given a
- * devicetree node identifier, a property specifying a GPIO and an index.
+ * devicetree node identifier, a property specifying a GPIO and an index. If the
+ * given @p node_id is not found or not enabled, this returns an empty
+ * initializer.
```

Comment:
I've got some whiplash here. Elsewhere you're all for adding OR_NULL and deprecating ANY, and here you seem to be going back to silently falling back on null expansions

Reply:
As said above, if this move forward, we should just have one set of macros: `DEVICE_DT_GET` and DEVICE_DT_GET_ANY`. Both may return NULL, compile time error is moved to the declaration macros. No more ONE/OR_NULL discussions :-)